crates/bevy_ecs/src/query/access.rs

```diff
@@ -325,6 +318,84 @@ impl<T: SparseSetIndex> FilteredAccess<T> {
     }
 }
 
+// A struct to express something like `Or<(With<A>, With<B>)>`.
+// Filters like `(With<A>, Or<(With<B>, With<C>)>` are expanded into `Or<((With<A>, With<B>), (With<A>, With<C>))>`.
```

Comment:
I'd like to see better docs explaning the roles of the generics. It's mentioned below, but it can be hard to follow when you're encountering this type for the first time.

Reply:
I've expanded the docs a bit, let me know if that works

Reply:
got rid of the additional generic argument eventually